subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -529,242 +600,210 @@ func (h *HivedAlgorithm) findVirtualCell(
-func allocatePhysicalInsidePreassigned(preassignedPhysical *PhysicalCell, vcCell *VirtualCell) *PhysicalCell {
-	// traverse the cell hierarchy to find the highest-level virtual cell
-	// that has not been bound to any physical cell
-	var boundCell *PhysicalCell
-	for c := vcCell; ; {
-		if c.GetPhysicalCell() != nil {
-			boundCell = c.GetPhysicalCell()
-			break
-		} else if c == vcCell.preAssignedCell {
-			break
-		} else {
-			c = c.GetParent().(*VirtualCell)
+		if !common.StringsContains(suggestedNodes, selectedNode) {
+			panic(fmt.Sprintf("[%v]: node %v picked by algorithm but not in K8S candidates",
+				internal.Key(pod), selectedNode))
 		}
-	}
-	if boundCell == nil {
-		boundCell = preassignedPhysical
-	}
-
-	for boundCell.GetLevel() > vcCell.GetLevel() {
-		boundCell = selectCellFromBuddies(boundCell.GetChildren()).(*PhysicalCell)
-	}
-	if boundCell.GetPriority() > opportunisticPriority {
-		panic(fmt.Sprintf("Cannot find physical cell for %v", vcCell.GetName()))
-	}
-	return boundCell
-}
-
-// getCellViewWithPriority returns a free cell list of a chain given a priority.
-// A cell will be invisible if it is with a equal-or-higher priority, or all of its buddies and itself
-// are with lower priorities. In the latter case, all of the buddies are masked and effectively "merged".
-func getCellViewWithPriority(fullCellList ChainCellList, p CellPriority) ChainCellList {
-	top := CellLevel(len(fullCellList))
-	cellView := NewChainCellList(top)
-	for l := CellLevel(1); l <= top; l++ {
-		for _, c := range fullCellList[l] {
-			if c.GetPriority() < p &&
-				(c.GetParent() == nil || maxPriority(c.GetParent().GetChildren()) >= p) {
-				cellView[l] = append(cellView[l], c)
+		if preemptionVictims.IsEmpty() {
+			klog.Infof("[%v]: scheduled to node %v, GPUs %v",
+				internal.Key(pod), selectedNode, selectedGpuIndices)
+			return internal.PodScheduleResult{
+				PodBindInfo: &api.PodBindInfo{
+					Node:                  selectedNode,
+					GpuIsolation:          selectedGpuIndices,
+					CellChain:             chain,
+					AffinityGroupBindInfo: gsi,
+				},
 			}
-		}
-	}
-	return cellView
-}
-
-// getCellViewForPodPlacement returns a free cell list inside a physical cell,
-// where a cell is free if it has no pod running (on itself or its children).
-// Similar to getCellViewWithPriority, buddy cells that are all free will be invisible ("merged").
-// This function is used for pod placement with buddy alloc.
-func getCellViewForPodPlacement(fullCellList ChainCellList) ChainCellList {
-	top := CellLevel(len(fullCellList))
-	cellView := NewChainCellList(top)
-	for l := CellLevel(1); l <= top; l++ {
-		for _, c := range fullCellList[l] {
-			cc := c.(*PhysicalCell)
-			if !cc.HasPod() {
-				if p := cc.GetParent(); p == nil || l == top || p.(*PhysicalCell).HasPod() {
-					cellView[l] = append(cellView[l], cc)
-				}
+		} else {
+			return internal.PodScheduleResult{
+				PodPreemptInfo: &internal.PodPreemptInfo{},
```

Comment:
>PodPreemptInfo: &internal.PodPreemptInfo{}, [](start = 4, length = 43)

After you returns victims for the specific node, please replace my previous comment:
 // Including all victim Pods in the victim cell, although the default scheduler
 // will only preempt victims on one node.

to be

 // Only need to include the victim Pods of current preemptor Pod.